src/ir_Metz.cpp

```diff
@@ -0,0 +1,100 @@
+/// @param[in] repeat Nr. of times the message is to be repeated.
+void IRsend::sendMetz(const uint64_t data, const uint16_t nbits,
+                      const uint16_t repeat) {
+  sendGeneric(kMetzHdrMark, kMetzHdrSpace,       // Header
+              kMetzBitMark, kMetzOneSpace,       // Data
+              kMetzBitMark, kMetzZeroSpace,
+              kMetzBitMark, kDefaultMessageGap,  // Footer.
+              data, nbits,                       // Payload
+              kMetzFreq, true, repeat, kDutyDefault);
+}
+
+/// Encode a Metz address, command, and toggle bits into a code suitable
+/// for use with sendMetz().
+/// @param[in] address A 6-bit address value.
+/// @param[in] command A 3-bit command value.
+/// @param[in] toggle Should the toggle bit be set in the result?
+/// @return A 19-bit value suitable for use with `sendMetz()`.
+uint32_t IRsend::encodeMetz(const uint8_t address, const uint8_t command,
+                            const bool toggle) {
+  return toggle << (2 * (kMetzAddressBits + kMetzCommandBits)) |
+      (command & 0x7) << (2 * kMetzAddressBits + kMetzCommandBits) |
+      (~command & 0x7) << (2 * kMetzAddressBits) |
+      (address & 0x3F) << kMetzAddressBits |
+      (~address & 0x3F);
+}
+#endif  // SEND_METZ
+
+#if DECODE_METZ
+/// Decode the supplied Metz message.
+/// Status: BETA / Probably works.
+/// @param[in,out] results Ptr to the data to decode & where to store the decode
+/// @param[in] offset The starting index to use when attempting to decode the
+///   raw data. Typically/Defaults to kStartOffset.
+/// @param[in] nbits The number of data bits to expect.
+/// @param[in] strict Flag indicating if we should perform strict matching.
+/// @return A boolean. True if it can decode it, false if it can't.
+bool IRrecv::decodeMetz(decode_results *results, uint16_t offset,
+                        const uint16_t nbits, const bool strict) {
+  if (strict && nbits != kMetzBits) return false;
+
+  uint64_t data = 0;
+
+  // Match Header + Data + Footer
+  if (!matchGeneric(results->rawbuf + offset, &data,
+                    results->rawlen - offset, nbits,
+                    kMetzHdrMark, kMetzHdrSpace,       // Header
+                    kMetzBitMark, kMetzOneSpace,       // Data
+                    kMetzBitMark, kMetzZeroSpace,
+                    kMetzBitMark, kDefaultMessageGap,  // Footer
+                    true, _tolerance, 0, true)) return false;
+
+  uint16_t address = GETBITS64(data, kMetzAddressBits, kMetzAddressBits);
+  uint16_t command = GETBITS64(data, 2 * kMetzAddressBits + kMetzCommandBits,
+                               kMetzCommandBits);
+  // Compliance
+  if (strict) {
+    if (address != invertBits(GETBITS64(data, 0, kMetzAddressBits),
+                              kMetzAddressBits) ||
+        command != invertBits(GETBITS64(data, 2 * kMetzAddressBits,
+                                                 kMetzCommandBits),
```

Comment:
Indentation?

Reply:
Thanks! Took me a second to see it. Fixed!